lesson11/main.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const body = document.getElementById("body");
+const wrapper = document.createElement("div");
+body.appendChild(wrapper);
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+const endLoading = () => wrapper.remove();
+const jsonURL = "https://jsondata.okiba.me/v1/json/KKl45211103113313";
+
+async function fetchData() {
+   try {
+      const response = await fetch(jsonURL);
+      const json = await response.json();
+      return json.data;
+   } catch(error) {
+      throw new Error(`データの取得に失敗しました：${error}`);
+   } finally {
+      console.log('処理が終了しました。');
+   }
+}
+
+async function init() {
+   startLoading();
+   try {
+      const result = await fetchData();
+      createListElements(result);
+   } catch(e) {
+      console.log('データが取得できませんでした');
+      console.error(e);
```

Comment:
How about using the same process as fetchData here?

In its current state.
If the fetchData succeeds, but the data has an unexpected shape, an error will occur because the "data.length;" cannot be checked.

It would be nice to do the same (`データの取得に失敗しました：${error}`) in that case.

Reply:
Thank you for your review.
Is my fix appropriate to your point?
I would appreciate it if you confirm.

82f166f

Reply:
I think it would be better to use the "e" argument of catch (as in fetchData).
If an error occurs in fetchData, the catch(e) in init() will contain the error thrown, and we want to display it.
For any other errors that occur, I want to display the details of the error and the fact that an error occurred.

When an error occurs in fetchData, the catch(e) in init() will contain the error that was thrown, and I want to display it.


I've written down my thoughts.
[codesandbox](https://codesandbox.io/s/unruffled-euclid-buosp?file=/main.js)

Reply:
Thank you for your detailed comments.
I was impressed to see the smartly handled code for the two patterns of errors.
I learned a lot from it. I would like to get into the habit of writing code with possible errors in mind.
I fixed that point, so please check.

10d6dc2